samples/batch-process-document.v1beta3.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2020 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const uuid = require('uuid');
+
+async function main(
+  projectId = 'YOUR_PROJECT_ID',
+  location = 'YOUR_PROJECT_LOCATION',
+  processorId = 'YOUR_PROCESSOR_ID', // Create this in the Cloud Console
+  gcsInputUri = 'gs://cloud-samples-data/documentai/invoice.pdf',
+  gcsOutputUri = 'output-bucket',
+  gcsOutputUriPrefix = uuid.v4()
+) {
+  // [START documentai_batch_process_document]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const projectId = 'YOUR_PROJECT_ID';
+  // const location = 'YOUR_PROJECT_LOCATION'; // Format is 'us' or 'eu'
+  // const processorId = 'YOUR_PROCESSOR_ID';
+  // const gcsInputUri = 'YOUR_SOURCE_PDF';
+  // const gcsOutputUri = 'YOUR_STORAGE_BUCKET';
+  // const gcsOutputUriPrefix = 'YOUR_STORAGE_PREFIX';
+
+  // Imports the Google Cloud client library
+  const {
+    DocumentProcessorServiceClient,
+  } = require('@google-cloud/documentai').v1beta3;
+  const {Storage} = require('@google-cloud/storage');
+
+  const clientOptions = {
+    apiEndpoint: 'us-documentai.googleapis.com',
```

Comment:
Interesting!  Why do we need to specify this in the client options?  I'd expect it to be available as the default.

Reply:
My _understanding_ is that this is the new way of doing things. According to Cloud AI, all new APIs are supposed to specify a location-specific hostname.

Reply:
Yes, my understanding is this is the new normal (at least according to our product team folks). Customers need to specify a location with their hostname for any call to the API.

Reply:
What happens if you don't include the location?  I am adding @alexander-fenster and @jskeet because this is kind of a big deal.

Reply:
There was definitely talk a while ago about this, but I thought there'd been pushback on it being required. Let's take the discussion of that to internal email.

Reply:
@JustinBeckwith If you don't include the location then your requests fail. There is no global / default location set for this service.

Reply:
Actually you know what, this service has a default `us-documentai.googleapis.com` hostname. I have removed the `clientOptions` code as a result.